core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/DatalakeV4Controller.java

```diff
@@ -28,11 +31,46 @@ public class DatalakeV4Controller implements DatalakeV4Endpoint {
     @Inject
     private WorkspaceService workspaceService;
 
+    @Inject
+    private DiskUpdateService diskUpdateService;
+
     @Override
+    @InternalOnly
     public StackViewV4Responses list(@Deprecated String environment, @TenantAwareParam String environmentCrn) {
         if (StringUtils.isBlank(environmentCrn) || StringUtils.isNotBlank(environment)) {
             throw new BadRequestException("environment param is deprecated, only environmentCrn should be filled.");
         }
         return stackOperations.listByEnvironmentCrn(workspaceService.getForCurrentUser().getId(), environmentCrn, List.of(StackType.DATALAKE));
     }
+
+    @Override
+    public Boolean isDiskTypeChangeSupported(String platform) {
+        if (StringUtils.isBlank(platform)) {
+            throw new BadRequestException("Platform param is null or empty.");
+        }
+        return diskUpdateService.isDiskTypeChangeSupported(platform);
+    }
+
+    @Override
+    public void updateDiskTypeAndSize(SdxDiskUpdateRequest sdxDiskUpdateRequest) throws Exception {
+        DiskUpdateRequest diskUpdateRequest = sdxDiskUpdateRequest.getDiskUpdateRequest();
+        List<VolumeSetAttributes.Volume> volumesToUpdate = sdxDiskUpdateRequest.getVolumesToUpdate();
+        String cloudPlatform = sdxDiskUpdateRequest.getCloudPlatform();
+        diskUpdateService.updateDiskTypeAndSize(diskUpdateRequest, volumesToUpdate, cloudPlatform);
+    }
+
+    @Override
+    public void stopCluster(long stackId) throws Exception {
+        diskUpdateService.stopCluster(stackId);
+    }
+
+    @Override
+    public void startCluster(long stackId) throws Exception {
+        diskUpdateService.startCluster(stackId);
+    }
+
+    @Override
+    public void resizeDisks(long stackId, String instanceGroup) throws Exception {
+        diskUpdateService.resizeDisks(stackId, instanceGroup);
```

Comment:
No - the main logic is in the service layer - which is what can be reused for DH. that is the reason i added the logic to a generic service layer and added the endpoint specific for DL so it can be accessed from DL

Reply:
Then create please a new controller for disk updates and put these codes into there.